src/Workspaces/SharedUtilitiesAndExtensions/Workspace/CSharp/LanguageServices/CSharpRemoveUnnecessaryImportsService.Rewriter.cs

```diff
@@ -48,6 +47,8 @@ public override SyntaxNode DefaultVisit(SyntaxNode node)
                 out SyntaxTriviaList finalTrivia)
             {
                 var currentUsings = new List<UsingDirectiveSyntax>(usings);
+                var onlyDeletedUsingsBefore = true;
```

Comment:
This name doesn't make much sense. It starts off true, but we haven't deleted any usings yet.

Reply:
I'm struggling with naming here. In my first commit I called it `firstUsingGroup`, but it is not a good name either IMO. This flag is here to determine the first using that doesn't need to be deleted, e.g.:
```csharp
using Namespace1; //deleted
using Namespace2; // deleted
using Namespace3; // no need to delete. 'onlyDeletedUsingsBefore' is true, so perform logic of possible blank lines trimming
using Namespace4; // no need to delete. 'onlyDeletedUsingsBefore' is false, do not apply any additional logic
```
So naming suggestions are welcome!

Reply:
What you just wrote sounds fine to me: `firstUsingNotBeingRemoved`.